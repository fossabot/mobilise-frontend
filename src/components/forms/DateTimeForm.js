import React from 'react';
import { Form, Col, Row } from 'react-bootstrap';
import moment from 'moment';

const DateTimeForm = props => {
  const { handleChange } = props;
  return (
    <Form.Row>
      {/* Date */}
      <Form.Group as={Col}>
        <Form.Label>Date</Form.Label>
        <Form.Control
          id="date"
          name="date"
          onChange={handleChange}
          type="date"
          min={moment().format('YYYY-MM-DD')}
          required
        />
      </Form.Group>
      {/* Time */}
      <Form.Group as={Col}>
        <Row>
          <Col>
            <Form.Label>Start</Form.Label>
            <Form.Control
              id="start-time"
              name="startTime"
              type="time"
              onChange={handleChange}
              required
            />
          </Col>
          <Col>
            <Form.Label>End</Form.Label>
            <Form.Control
              id="end-time"
              name="endTime"
              type="time"
              onChange={handleChange}
              required
            />
          </Col>
        </Row>
      </Form.Group>
    </Form.Row>
  );
};

export default DateTimeForm;
