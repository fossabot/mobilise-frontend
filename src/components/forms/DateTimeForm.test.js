import React from 'react';
import { shallow } from 'enzyme';
import DateTimeForm from './DateTimeForm';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<DateTimeForm />);
});

describe('Date and Time Form Component', () => {
  it('renders form correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
