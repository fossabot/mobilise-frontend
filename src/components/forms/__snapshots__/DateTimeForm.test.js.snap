// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date and Time Form Component renders form correctly 1`] = `ShallowWrapper {}`;
